src/semantic-ui/FileUpload.js

```diff
@@ -164,6 +166,9 @@ class FileUpload extends Component<Props, State> {
       }
     });
 
+    if (!(errors && errors.length) && this.props.onSuccess) {
```

Comment:
I think the logic behind how this component handles errors may have been flawed to begin with: The component calls `validate` to get the list of valid files, then calls `onFilesAdded` with those files and allows the implementing component to do whatever with those files (upload, render, etc).

Would it make more sense to _only_ call `onFilesAdded` if _all_ of the files pass validation?

Reply:
You're right. That makes sense to me!